File: mobile_ui/quickchat_ui/lib/features/chat/domain/repositories/chat_repository.dart
import 'package:dartz/dartz.dart';
import 'package:quickchat_ui/core/logic/result.dart';
import 'package:quickchat_ui/features/chat/data/dto/ConversationData.dto.dart';
import 'package:quickchat_ui/features/chat/data/dto/Conversations.dto.dart';
import 'package:quickchat_ui/features/chat/data/dto/ConversationsData.dto.dart';
import 'package:quickchat_ui/features/chat/data/dto/GetConversations.dto.dart';

abstract class ChatRepository {
  Stream<Result<ConversationsDto>> connectChatStream();
  void disposeChatStream();
  Future<Result<ConversationDataDto>> getConversation({
    required int page,
  });
  Future<Result<ConversationsDataDto>> getConversations({
    required GetConversationsDto getConversations,
  });
  Future<Result<Unit>> sendChat({
    required String recipient,
    required String content,
  });
}

File: web_ui/app/features/chat/domain/repositories/chat_repository.ts
import { Result } from "@/app/core/result";
import { ConversationDataDto } from "../../data/dto/ConversationData.dto";
import { ConversationsDataDto } from "../../data/dto/ConversationsData.dto";
import { ConversationsDto } from "../../data/dto/Conversations.dto";
import { Observable } from "rxjs";

export interface ChatRepository {
  connectChatStream(): Observable<Result<ConversationsDto>>;
  disposeChatStream(): Promise<Result<void>>;
  getConversation(page: number): Promise<Result<ConversationDataDto>>;
  getConversations(
    conversationsId: string,
    page: number
  ): Promise<Result<ConversationsDataDto>>;
  sendMessage(recipient: string, content: string): Promise<Result<boolean>>;
}
